tracing-subscriber/src/fmt/mod.rs

```diff
@@ -430,6 +430,24 @@ where
     }
 }
 
+#[derive(Debug)]
+enum InitError {
+    #[cfg(feature = "tracing-log")]
+    Log(tracing_log::log_tracer::SetLoggerError),
+    Dispatcher(tracing_core::dispatcher::SetGlobalDefaultError),
+}
+
+impl fmt::Display for InitError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            InitError::Log(err) => err.fmt(f),
+            InitError::Dispatcher(err) => err.fmt(f),
+        }
+    }
+}
+
+impl Error for InitError {}
```

Comment:
This doesn't implement `Error::cause`, which (since it's composed of two `Error`s) it probably should. Unfortunately, implementing `cause` would require the `log` error to implement `Error`, which puts us back at square one. We should probably just make sure that the `log/std` feature is enabled, instead, and go back to returning a box.

Reply:
Done